File: algorithms/100161.compress_string_lcci/compress_string_lcci.rs
impl Solution {
    pub fn compress_string(s: String) -> String {
        if s == "" {
            return s;
        }
        let mut chars = String::new();
        let (mut ch, mut count) = (s.as_bytes()[0] as char, 0);
        for c in s.chars() {
            if ch != c {
                chars.push(ch);
                chars.push_str(&count.to_string());
                ch = c;
                count = 1;
            } else {
                count += 1
            }
        }
        chars.push(ch);
        chars.push_str(&count.to_string());
        return if s.len() > chars.len() { chars } else { s };
    }
}

File: algorithms/100161.compress_string_lcci/compress_string_lcci.cpp
class Solution {
public:
    string compressString(string &s) {
        s.push_back('#');
        string ans;
        size_t i = 0, start = i;
        while (i < s.size()) {
            if (s[start] != s[i]) {
                ans.push_back(s[start]);
                ans += to_string(i - start);
                start = i;
            }
            ++i;
        }
        s.pop_back();
        return ans.size() < s.size() ? ans : s;
    }
};